samza-api/src/main/java/org/apache/samza/storage/TaskBackupManager.java

```diff
@@ -60,10 +61,12 @@ default void start(Checkpoint checkpoint) {}
    * @param stateCheckpointMarkers The map of storename to checkpoint makers returned by the snapshot
    * @return The future of storename to checkpoint map of the uploaded local store
    */
-  CompletableFuture<Map<String, StateCheckpointMarker>> upload(CheckpointId checkpointId, Map<String, StateCheckpointMarker> stateCheckpointMarkers);
+  CompletableFuture<Map<String, StateCheckpointMarker>> upload(CheckpointId checkpointId,
+      Map<String, StateCheckpointMarker> stateCheckpointMarkers);
 
   /**
-   * Persist the state locally to the file system
+   * Occurs after the state has been persisted. Writes a copy of the persisted StateCheckpointMarkers from
```

Comment:
Should this be called independently by each storage backend (resulting in many different "OFFSET-V2" equivalent files), or should this be called once by TaskInstance / TaskCommitManager after collecting all the inputOffsets and SCMs and creating the checkpoint?